packages/unified-selection/src/unified-selection/iModel/EnableUnifiedSelectionSyncWithIModel.ts

```diff
@@ -0,0 +1,46 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+/** @packageDocumentation
+ * @module UnifiedSelection
+ */
+
+import { CachingHiliteSetProvider } from "../CachingHiliteSetProvider";
+import { IMetadataProvider } from "../queries/ECMetadata";
+import { IECSqlQueryExecutor } from "../queries/ECSqlCore";
+import { ElementSelectionScopeProps, SelectionScope } from "../SelectionScope";
+import { SelectionStorage } from "../SelectionStorage";
+import { IModelConnection } from "./IModel";
+import { IModelSelectionHandler } from "./IModelSelectionHandler";
+
+/**
+ * Props for `enableUnifiedSelectionSyncWithIModel`.
+ * @internal Not exported through barrel, but used in public API as an argument. May be supplemented with optional attributes any time.
+ */
+export interface EnableUnifiedSelectionSyncWithIModelProps {
+  /** iModel to synchronize selection for. */
+  iModel: IModelConnection;
```

Comment:
It seems really weird that we require `IModelConnection` and, separately - IModel's query executor and metadata provider. I think `IModelConnection` needs to be renamed to reflect it's purpose more closely and not confuse consumers into thinking that this is a real `IModelConnection`.

Reply:
Can't say I love the name either. After discussing the topic with @saskliutas, we came up with a plan to refactor the way we access imodels in our new packages. So let's leave this as it is now and refactor that afterwards.